### PR TITLE
[FORM] Text Field

### PR DESCRIPTION
**Pull request by epifaniovillaro**
## Issue Link
[LMS-317](https://framgiaph.backlog.com/view/LMS-317)

## Steps to reproduce
cd client, npm run dev
test the component : http://localhost:3000/demo/components
location of the component: shared/components/Text-field/text-field.tsx
## Affected Components / Functionalities / Page
demo/components/index.tsx

## Test Cases
_will update soon..._

## Notes
n/a

## Screenshots
image
image

### CODE REVIEW

**Review by Marielleregner7 — APPROVED** [review]
**APPROVED**

- Able to create an Identical to the wireframe Text field in the page and can be seen in the http://localhost:3000/demo/components/Text-field

Comments: 0
Test time: 17 mins
Iterations: 0

**Comment by epifaniovillaro** [comment]
image

File: client/src/pages/demo/components/index.tsx [hunk 1]
```diff
@@ -29,6 +30,30 @@ const DemoComponent: React.FunctionComponent = () => {
           </div>
         </div>
 
+        <div className="p-4 border" id='text-field'>
+          <Textfield
+            label = "Description"
+            placeholder= "Enter a description"
+            width = {50}
+            height= {10}
+            fontsize = "text-xl"
```
Comment (escbooster12-dev) [inline client/src/pages/demo/components/index.tsx:39]: - can you pass `className` prop instead of `fontsize`

File: client/src/shared/components/Text-field/text-field.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  fontsize: string;
+  resizable: boolean;
+}
+
+const Textfield: React.FC<Props> = ({ label, placeholder, width, height, fontsize, resizable }: Props) => {
+  return (
+          <div className="p-5 h-auto w-auto">
+            <div>
+              <label>{label} </label>
+            </div>
```
Comment (escbooster12-dev) [inline client/src/shared/components/Text-field/text-field.tsx:17]: - `label` prop should be optional
- if no `label` being defined, it should not render the `div` and an empty `label` html tag

File: client/src/shared/components/Text-field/text-field.tsx [hunk 3]
```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+
+interface Props {
+  label: string;
+  placeholder: string;
```
Comment (escbooster12-dev) [inline client/src/shared/components/Text-field/text-field.tsx:5]: - `placeholder` should be optional

File: client/src/shared/components/Text-field/text-field.tsx [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+            <div>
+              <label>{label} </label>
+            </div>
+            <textarea
+            rows = {height}
+            cols = {width}
+              className={`${
+                resizable ? 'resize' : 'resize:none'
+              } border border-black rounded-lg rows`}
+              placeholder={`${placeholder}`}
```
Comment (escbooster12-dev) [inline client/src/shared/components/Text-field/text-field.tsx:24]: - do not use backticks if the appended content is just one.
```suggestion
              placeholder={placeholder}
```

**Comment by epifaniovillaro** [comment]
image

File: client/src/shared/components/Text-field/text-field.tsx [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+  width: number;
+  height: number;
+  className: string;
+  resizable: boolean;
+}
+
+const Textfield: React.FC<Props> = ({ label, placeholder, width, height, className, resizable }: Props) => {
+  return (
+          <div className="p-5 h-auto w-auto">
+            {label !== '' ? (<div><label>{label} </label></div>) : null}
```
Comment (escbooster12-dev) [inline client/src/shared/components/Text-field/text-field.tsx:15]: - it will be more readable pag naka multiline.
- as much as possible limit empty conditions by using conditional operators

```suggestion
            {label !== null && (
                 <div>
                     <label>{label}</label>
                 </div>
             )}
```

File: client/src/shared/components/Text-field/text-field.tsx [hunk 6]
```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+
+interface Props {
+  label?: string;
+  placeholder?: string;
```
Comment (escbooster12-dev) [inline client/src/shared/components/Text-field/text-field.tsx:5]: - please add default props for `label` and `placeholder`. empty string is okay
- please run prettier in the file. I think it's not well formatted.

**Comment by epifaniovillaro** [comment]
image

File: client/src/pages/demo/components/index.tsx [hunk 7]
```diff
@@ -22,11 +23,9 @@ const DemoComponent: React.FunctionComponent = () => {
         <hr className="w-3/4 mx-auto" />
 
         <div className="p-4 border">
-          <h1>Component: Iframe</h1>
-          <br />
           <Iframe
             src="https://www.youtube.com/embed/HGl75kurxok"
-            className="border"
+            className={''}
```
Comment (escbooster12-dev) [inline client/src/pages/demo/components/index.tsx:28]: - i'm not sure. whats the changes for.
- please double check _"file changes"_ tab in your PR to fix possible overlooked changes.

File: client/src/pages/demo/components/index.tsx [hunk 8]
```diff
@@ -63,6 +62,96 @@ const DemoComponent: React.FunctionComponent = () => {
...
+          <div className="mt-[5px]">
+            <h1>Props: Text-field</h1>
+
+            <div className="bg-gray-300 p-[5px]">
+              <p> label = (string) ex. label="Description"</p>
+              <p> placeholder = (string)ex. placeholder="Enter Description"</p>
+              <p> width = (number) ex. width=10 </p>
+              <p> height = (number)ex. height=10 </p>
+              <p>
+                {' '}
```
Comment (escbooster12-dev) [inline client/src/pages/demo/components/index.tsx:144]: - what's this empty string for? :bow:

File: client/src/pages/demo/components/index.tsx [hunk 9]
```diff
@@ -63,6 +62,96 @@ const DemoComponent: React.FunctionComponent = () => {
...
+                fontsize = (string)ex. fontsize="text-s" | fontsize="text-2xl"
+              </p>
+              <p>
+                {' '}
```
Comment (escbooster12-dev) [inline client/src/pages/demo/components/index.tsx:148]: - also this one.

File: client/src/shared/components/Text-field/text-field.tsx [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+      <textarea
+        rows={height}
+        cols={width}
+        className={`${resizable ? 'resize' : 'resize:none'} ${className}`}
+        placeholder={placeholder}
+      ></textarea>
+    </div>
+  );
+};
+Textfield.defaultProps = {
```
Comment (escbooster12-dev) [inline client/src/shared/components/Text-field/text-field.tsx:36]: - add newline
```suggestion
};

Textfield.defaultProps = {
```

File: client/src/shared/components/Text-field/text-field.tsx [hunk 11]
```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+
```
Comment (escbooster12-dev) [inline client/src/shared/components/Text-field/text-field.tsx:2]: - filename should be `index.tsx`
image

**Comment by epifaniovillaro** [comment]
image

File: client/src/shared/components/Text-field/index.tsx [hunk 12]
```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+
+interface Props {
```
Comment (escbooster12-dev) [inline client/src/shared/components/Text-field/index.tsx:3]: - this interface could be usable in the future development. so please export it.
```suggestion
export interface Props {
```

File: client/src/shared/components/Text-field/index.tsx [hunk 13]
```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+
```
Comment (escbooster12-dev) [inline client/src/shared/components/Text-field/index.tsx:2]: - for shared file naming convention. please use _Pascal case_
image

File: client/src/pages/demo/components/index.tsx [hunk 14]
```diff
@@ -22,12 +23,7 @@ const DemoComponent: React.FunctionComponent = () => {
         <hr className="w-3/4 mx-auto" />
 
         <div className="p-4 border">
-          <h1>Component: Iframe</h1>
-          <br />
-          <Iframe
-            src="https://www.youtube.com/embed/HGl75kurxok"
-            className="border"
-          />
+          <Iframe src="https://www.youtube.com/embed/HGl75kurxok" />
```
Comment (escbooster12-dev) [inline client/src/pages/demo/components/index.tsx:26]: - unnecessary changes. please undo this to original state.

**Comment by epifaniovillaro** [comment]
image

File: client/src/shared/components/TextField/index.tsx [hunk 15]
```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+
+export interface Props {
+  label?: string;
+  placeholder?: string;
+  width: number;
+  height: number;
+  className: string;
```
Comment (escbooster12-dev) [inline client/src/shared/components/TextField/index.tsx:8]: - className is not required. Empty string is okay as default value.

**Comment by kdacudag** [comment]
image

**Review by Marielleregner7 — APPROVED** [review]
**APPROVED**

- Text field is visible in the page and can be used dynamically. 
- No ESLint warnings or errors.

Comments: 22
Test time: 38 mins
Iterations: 5
